File: JavaScript/8js 2/nearestSq.js
/*
|--------------------------------------------------------------------------
| Find Nearest square number
|--------------------------------------------------------------------------
|
*/

function nearestSq(n) {
   return Math.round(Math.sqrt(n)) ** 2;
}


function nearestSq(n) {
   return Math.round(n ** 0.5) ** 2;
}


function nearestSq(n) {
   return Math.pow(Math.round(Math.sqrt(n)), 2);
}


console.log(nearestSq(1), 1);
console.log(nearestSq(2), 1);
console.log(nearestSq(10), 9);
console.log(nearestSq(111), 121);
console.log(nearestSq(9999), 10000);

File: Python/8py 2/nearest_sq.py
"""
|--------------------------------------------------------------------------
| Find Nearest square number
|--------------------------------------------------------------------------
|
"""

def nearest_sq(n):
   return round(n**0.5)**2


nearest_sq = lambda n: round(n ** 0.5) ** 2


from math import sqrt
def nearest_sq(n):
   return pow(round(sqrt(n)), 2)


print(nearest_sq(1), 1)
print(nearest_sq(2), 1)
print(nearest_sq(10), 9)
print(nearest_sq(111), 121)
print(nearest_sq(9999), 10000)